python/lsst/meas/base/afterburner.py

```diff
@@ -0,0 +1,183 @@
+from collections import namedtuple
+import sys
+import traceback
+
+import lsst.pipe.base
+import lsst.pex.config
+import lsst.daf.base
+
+from .pluginsBase import BasePlugin, BasePluginConfig
+from .pluginRegistry import PluginRegistry
+from .baseLib import FatalAlgorithmError, MeasurementError
+
+# Exceptions that the measurement tasks should always propagate up to their callers
+FATAL_EXCEPTIONS = (MemoryError, FatalAlgorithmError)
```

Comment:
What about `SystemExit` and `KeyboardInterrupt`?  I think you only want to catch things that inherit from `Exception` (not `BaseException`).